src/lib.rs

```diff
@@ -196,33 +203,42 @@ impl State {
             Backend::BrowserWebGpu => "WebGPU",
         };
 
-        let (device, queue) = adapter
-            .request_device(
-                &wgpu::DeviceDescriptor {
-                    features: wgpu::Features::empty(),
-                    limits: wgpu::Limits::downlevel_webgl2_defaults(),
-                    label: None,
+        let (device, queue) = adapter.request_device(
+            &wgpu::DeviceDescriptor {
+                features: wgpu::Features::empty(),
+                limits: if cfg!(target_arch = "wasm32") {
+                    wgpu::Limits::downlevel_webgl2_defaults()
```

Comment:
I've added a WebGPU version now, so don't assume that wasm will always be webgl2 (you can check for the `webgl` feature instead)